src/rdkafka_idempotence.c

```diff
@@ -145,38 +145,52 @@ rd_kafka_idemp_broker_any (rd_kafka_t *rk,
  * @brief Check if an error needs special attention, possibly
  *        raising a fatal error.
  *
+ * @param is_fatal if true, force fatal error regardless of error code.
+ *
  * @returns rd_true if a fatal error was triggered, else rd_false.
  *
  * @locks rd_kafka_wrlock() MUST be held
  * @locality rdkafka main thread
  */
 rd_bool_t rd_kafka_idemp_check_error (rd_kafka_t *rk,
```

Comment:
the passing in of `is_fatal` as well as processing in this function to possibly update it is a bit conflated. maybe have a function that does only does the stuff out of `if (!is_fatal) return rd_false;` and if the stuff before is needed in multiple places, have that in another function.

Reply:
i guess it makes the preface logic harder... whatever you like, i think the point holds though